### DISCUSSION TITLE
show bookmarks view in side bar automatically

### DISCUSSION THREAD

**Discussion by kumar007git** [alefragnani/vscode-bookmarks#576] [Q&A] [answered]
Hi,
How can I make VSCode show bookmarks view in the sidebar automatically if my file has bookmarks in it?

One a side not, how can I define default view in the sidebar? For example, whenever I open VSCode right now it shows the Workspace view but how can I make it show bookmarks view by default?

Thx.

**Comment by alefragnani** [comment, marked as answer]
Hi @kumar007git ,

I’m not aware of any VS Code setting that defines such behavior, nor any API that could be used by an extension  to _force_ its view/side bar when some project is opened/created.

On the other hand, VS Code stores the _state_ of the workspace, and when you _return_ to that workspace, it should reopen the same views/editors/panels you had before.

I’ll take a closer look at VS Code API in order to locate a way to automatically display some view when opening a project. If there is one, I could create a setting for such behavior. I’ll post updates here whenever possible.

Hope this helps
